### PR TITLE
Remove TypeUuid

### PR DESCRIPTION
**Pull request by AxiomaticSemantics**
# Objective
TypeUuid is deprecated, remove it.

## Migration Guide
Convert any uses of `#[derive(TypeUuid)]` with `#[derive(TypePath]` for more complex uses see the relevant [documentation](https://docs.rs/bevy/latest/bevy/prelude/trait.TypePath.html) for more information.

### CODE REVIEW

File: crates/bevy_pbr/src/wireframe.rs
Comment (AxiomaticSemantics) [inline crates/bevy_pbr/src/wireframe.rs:198]: This uuid should also be removed correct?

**Comment by AxiomaticSemantics** [comment]
I have ran into one issue with the tests failing due to import conflicts and attempted to use `#[rustfmt=skip]` to prefix serde import with as `::serde` to no avail.

**Review by james7132 — APPROVED** [review]
I don't like that we're removing this without marking it deprecated first, but I'm OK with removing it cold-turkey as there should be nothing interacting with it anymore.

File: crates/bevy_reflect/src/lib.rs
Comment (MrGVSV) [inline crates/bevy_reflect/src/lib.rs:540]: Could you elaborate why this is needed?

**Review by MrGVSV — APPROVED** [review]
Would have been nice to have first made it `#[deprecated]`, but I don't think the migration should be too bad in most cases— assuming the user isn't already on 0.12.

**Comment by MrGVSV** [comment]
> Convert any uses of TypeUuid to use TypePath

@AxiomaticSemantics Could you update the Migration Guide to give users a bit more details on how this is done? Maybe a before/after snippet?

File: crates/bevy_pbr/src/wireframe.rs
Comment (soqb) [reply crates/bevy_pbr/src/wireframe.rs]: yup.

File: crates/bevy_reflect/src/lib.rs
Comment (AxiomaticSemantics) [reply crates/bevy_reflect/src/lib.rs:540]: This one probably isn't, adding glam as a dev-dep should take care of it, I'll remove it.

**Comment by AxiomaticSemantics** [comment]
> Would have been nice to have first made it `#[deprecated]`, but I don't think the migration should be too bad in most cases— assuming the user isn't already on 0.12.

I asked on discord, you thought it was, I was going to ask about this, nothing referenced it, but it wasn't marked as such. Up to SME/maintainers I guess, seems nothing in the previous guides mention it migrating away from it in favour of TypePath.

**Comment by AxiomaticSemantics** [comment]
> I don't like that we're removing this without marking it deprecated first, but I'm OK with removing it cold-turkey as there should be nothing interacting with it anymore.

If you'd rather mark it for now, this can hold off until release.

**Comment by alice-i-cecile** [comment]
My preference is to remove it outright: there's nothing connected to it.

**Review by alice-i-cecile — COMMENTED** [review]
What's up with the `#[rustfmt::skip]`? Are they needed? I have a pretty strong preference to remove them unless there's a good reason to have them.

**Comment by AxiomaticSemantics** [comment]
> What's up with the `#[rustfmt::skip]`? Are they needed? I have a pretty strong preference to remove them unless there's a good reason to have them.

I was getting import conflicts with serde and super::serde in the tests, the `::serde` prefix was to disambiguate. The annotation on glam I'll remove and test.

**Comment by soqb** [comment]
> > What's up with the `#[rustfmt::skip]`? Are they needed? I have a pretty strong preference to remove them unless there's a good reason to have them.
> 
> I was getting import conflicts with serde and super::serde in the tests, the `::serde` prefix was to disambiguate. The annotation on glam I'll remove and test.

i think i also had this at one point, can't remember the fix. are you getting the errors through rust-analyzer or through a cli command?

**Comment by AxiomaticSemantics** [comment]
> > > What's up with the `#[rustfmt::skip]`? Are they needed? I have a pretty strong preference to remove them unless there's a good reason to have them.
> > 
> > 
> > I was getting import conflicts with serde and super::serde in the tests, the `::serde` prefix was to disambiguate. The annotation on glam I'll remove and test.
> 
> i think i also had this at one point, can't remember the fix. are you getting the errors through rust-analyzer or through a cli command?

I noticed the errors becasuse bevy's CI was failing, no R/A in the mix here.

File: crates/bevy_reflect/src/lib.rs
Comment (AxiomaticSemantics) [reply crates/bevy_reflect/src/lib.rs]: This does seem to be required still, later comments cover this topic.

**Pull request merged**
